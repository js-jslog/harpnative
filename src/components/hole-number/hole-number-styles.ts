import { StyleSheet } from 'react-native'
import type { ViewStyle, TextStyle } from 'react-native'

import { sizes, colors } from '../../styles'

type HoleNumberStyles = {
  readonly cell: ViewStyle
  readonly text: TextStyle
}

const { 6: fontSize, 8: width } = sizes
const { holeNumbersColor } = colors

export const getStyles = (): HoleNumberStyles => {
  const styles = StyleSheet.create<HoleNumberStyles>({
    cell: {
      justifyContent: 'center',
      alignItems: 'center',
      width,
      height: 0,
    },
    text: {
      fontSize,
      color: holeNumbersColor,
    },
  })

  return styles
}
