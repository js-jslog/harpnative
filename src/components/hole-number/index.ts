export { HoleNumber } from './hole-number'
