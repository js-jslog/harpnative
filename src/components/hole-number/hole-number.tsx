import { Text, View } from 'react-native'
import React from 'react'
import type { ReactElement } from 'react'

import type { Coord } from '../../types'

export enum HoleNumberIds {
  One = '1',
  Two = '2',
  Three = '3',
  Four = '4',
  Five = '5',
  Six = '6',
  Seven = '7',
  Eight = '8',
  Nine = '9',
  Ten = '10',
  Eleven = '11',
  Twelve = '12',
  Thirteen = '13',
  Fourteen = '14',
  Fifteen = '15',
  Sixteen = '16',
}

type HoleNumberProps = {
  readonly xCoord: Coord
}

import { getStyles } from './hole-number-styles'

const styles = getStyles()

export function HoleNumber({ xCoord }: HoleNumberProps): ReactElement {
  const holeNumber: HoleNumberIds = `${xCoord + 1}` as HoleNumberIds

  return (
    <View style={styles.cell}>
      <Text style={styles.text}>{holeNumber}</Text>
    </View>
  )
}
