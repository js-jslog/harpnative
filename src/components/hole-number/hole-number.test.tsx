import React from 'react'
import { render } from '@testing-library/react-native'

import { HoleNumber, HoleNumberIds } from './hole-number'

test('HoleNumber renders a dom element with the expected value included', () => {
  const { getByText } = render(<HoleNumber xCoord={0} />)
  expect(getByText(HoleNumberIds.One)).toBeTruthy()
})

test('A snapshot of HoleNumber', () => {
  const { container } = render(<HoleNumber xCoord={0} />)
  expect(container).toMatchSnapshot()
})
