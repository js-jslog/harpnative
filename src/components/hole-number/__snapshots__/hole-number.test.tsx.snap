// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A snapshot of HoleNumber 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 0,
        "justifyContent": "center",
        "width": 54.433445391378655,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#495057",
          "fontSize": 20.79172382001754,
        }
      }
    >
      1
    </Text>
  </View>
</View>
`;
